packages/studio-be/src/core/migration/bot-migration-service.ts

```diff
@@ -68,58 +58,66 @@ export class BotMigrationService {
-{dim ${centerText(`${migrations.length} change${migrations.length === 1 ? '' : 's'}`, 40, 9)}}
-${_.repeat(' ', 9)}========================================`)
+    return missing.filter(x => x !== undefined) as MigrationFile[]
+  }
 
-    Object.keys(types).map(type => {
-      logger.warn(chalk`{bold ${types[type]}}`)
-      const filtered = migrations.filter(x => x.type === type)
+  private writeLabel(migrations: number, botId: string, version: string) {
+    const configLabel = `(${version} => ${this.migService.targetVersion})`
+    const changesLabel = `${migrations} change${migrations === 1 ? '' : 's'}`
 
-      if (filtered.length) {
-        filtered.map(x => logger.warn(`- ${x.description}`))
-      } else {
-        logger.warn('- None')
-      }
-    })
+    return chalk`{bold Migration${migrations === 1 ? '' : 's'} Required for ${botId}} ${configLabel} - ${changesLabel}`
   }
 
-  private async executeBotMigrations(botId: string, missingMigrations: MigrationFile[]) {
-    this.logger.info(chalk`
-${_.repeat(' ', 9)}========================================
-{bold ${centerText(
-      `Executing ${missingMigrations.length} migration${missingMigrations.length === 1 ? '' : 's'}`,
-      40,
-      9
-    )}}
-${_.repeat(' ', 9)}========================================`)
-
-    const opts = await this.migService.getMigrationOpts({ botId })
+  private async executeBotMigrations(botId: string, migrations: MigrationFile[], configVersion: string) {
+    const botConfig = await this.configProvider.getBotConfig(botId)
+    const opts = await this.migService.getMigrationOpts({ botId, botConfig })
     let hasFailures = false
 
-    await Promise.mapSeries(missingMigrations, async ({ filename }) => {
-      const result = await this.migService.loadedMigrations[filename].up(opts)
-      debug.forBot(botId, 'Migration step finished', { filename, result })
-      if (result.success) {
-        this.logger.forBot(botId).info(`- ${result.message || 'Success'}`)
-      } else {
-        hasFailures = true
-        this.logger.forBot(botId).error(`- ${result.message || 'Failure'}`)
+    const logger = this.logger.forBot(botId)
+
+    logger.warn(this.writeLabel(migrations.length, botId, configVersion))
+
+    await Promise.mapSeries(migrations, async ({ filename, info }) => {
+      const label = `${types[info.type]} - ${info.description}`
+
+      try {
+        const result = await this.migService.loadedMigrations[filename].up(opts)
+        debug.forBot(botId, 'Migration step finished', { filename, result })
+
+        if (result.success) {
+          logger.warn(`${chalk.green('[success]')} ${label}`)
+        } else if (result.hasChanges) {
+          logger.error(`${chalk.red('[failure]')} ${label}: ${result.message || ''}`)
+        }
+
+        return
+      } catch (err) {
+        logger.attachError(err).error(`${chalk.red('[failure]')} ${label}`)
       }
+
+      hasFailures = true
     })
 
+    logger.warn('')
```

Comment:
Is this voluntary?

Reply:
Yep, it adds a new line between migrations, if you had like 10 bots it would have been all cramped up

Reply:
Maybe add a comment so we don't remove it thinking that it was a mistake?

Reply:
I've changed the logic